src/js/components/PodInstancesTable.js

```diff
@@ -0,0 +1,334 @@
+    };
+
+    METHODS_TO_BIND.forEach(function (method) {
+      this[method] = this[method].bind(this);
+    }, this);
+  }
+
+  handleItemCheck(idsChecked) {
+    let checkedItems = {};
+
+    idsChecked.forEach(function (id) {
+      checkedItems[id] = true;
+    });
+    this.setState({checkedItems});
+  }
+
+  getColGroup() {
+    return (
+      <colgroup>
+        <col style={{width: '40px'}} />
+        <col />
+        <col style={{width: '120px'}} />
+        <col style={{width: '120px'}} />
+        <col style={{width: '64px'}} />
+        <col style={{width: '86px'}} />
+        <col style={{width: '160px'}} />
+        <col style={{width: '160px'}} />
+      </colgroup>
+    );
+  }
+
+  getColumnHeading(prop, order, sortBy) {
+    let caretClassNames = classNames(
+      'caret',
+      {
+        [`caret--${order}`]: order != null,
+        'caret--visible': prop === sortBy.prop
+      }
+    );
+
+    return (
+      <span>
+        {PodTableHeaderLabels[prop]}
+        <span className={caretClassNames}></span>
+      </span>
+    );
+  }
+
+  getColumnClassName(prop, sortBy, row) {
+    return classNames(`column-${prop}`, {
+      'highlight': prop === sortBy.prop,
+      'clickable': row == null,
+      'table-cell-task-dot': prop === 'status'
+    });
+  }
+
+  getColumns() {
+    return [
+      {
+        className: this.getColumnClassName,
+        heading: this.getColumnHeading,
+        prop: 'name',
+        render: this.renderColumnID,
+        sortable: true
+      },
+      {
+        className: this.getColumnClassName,
+        heading: this.getColumnHeading,
+        prop: 'address',
+        render: this.renderColumnAddress,
+        sortable: true
+      },
+      {
+        className: this.getColumnClassName,
+        heading: this.getColumnHeading,
+        prop: 'status',
+        render: this.renderColumnStatus,
+        sortable: true
+      },
+      {
+        className: this.getColumnClassName,
+        heading: this.getColumnHeading,
+        prop: 'cpus',
+        render: this.renderColumnResource,
+        sortable: true
+      },
+      {
+        className: this.getColumnClassName,
+        heading: this.getColumnHeading,
+        prop: 'mem',
+        render: this.renderColumnResource,
+        sortable: true
+      },
+      {
+        className: this.getColumnClassName,
+        heading: this.getColumnHeading,
+        prop: 'updated',
+        render: this.renderColumnUpdated,
+        sortable: true
+      },
+      {
+        className: this.getColumnClassName,
+        heading: this.getColumnHeading,
+        prop: 'version',
+        render: this.renderColumnVersion,
+        sortable: true
+      }
+    ];
+  }
+
+  getTableDataFor(instances) {
+    let podSpec = this.props.pod.getSpec();
+
+    return instances.getItems().map(function (instance) {
+      let resourcesSummary = {
+        cpus: 0, mem: 0, disk: 0, gpus: 0
+      };
+
+      let children = instance.getContainers()
```

Comment:
Can we do this nested map in a separate function? Then we can name it something that makes sense